src/osp/tasks/top_session.h

```diff
@@ -45,7 +45,13 @@
 #define OSP_AUX_DCDI_A(...) OSP_AUX_DCDI_B(OSP_AUX_DCDI_C(__VA_ARGS__))
 
 /**
- * @brief
+ * @brief Gives the next available section of topData to a session
+ * and declares an ID variable for each element that is given.
+ * The same ID's are also stored inside of the sessions m_data member
+ *
+ * @param arglist A variadic argument broken into two sections:
+ *     Firstly: 'count' The count is an integer equal to the number of id's
+ *     Secondly: id's Every id is a unique index into 'topData'
```

Comment:
instead of documenting implementation details like the macro taking a variadic argument list, just document what the macro takes.

sounds like the macro simply always requires a count parameter, so probably this OSP_DECLARE_CREATE_DATA_IDS() macros should be given the count parameter as an explicit parameter instead of relying on other macros that it call to break it out of the variadic argument.

Reply:
Nevermind this, the review was hiding some important details so i misread what was going on.